templates/service-account.yaml

```diff
@@ -2,7 +2,13 @@
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-    name: {{ template "neo4j.fullname" . }}-sa
+    {{- if .Values.serviceAccount.annotations }}
+    annotations:
+      {{-  range $key, $value := .Values.serviceAccount.annotations }}
+      {{ $key }}: "{{ $value }}"
+      {{- end }}
+    {{- end }}
+    name: {{ tpl .Values.serviceAccount.name . }}
```

Comment:
I think it's better to stick with the previous name which was removed by your diff.  I'd prefer to keep the name the same so that it's predictable on every deploy -- and additionally, since you're already annotating the service account above, you can keep .Values.serviceAccount.name and just put it into another annotation.

This would preserve the resource deploy footprint to be consistently named, and allow you to express what you need in annotations.  By contrast, using `name: {{ tpl .Values.serviceAccount.name . }}` isn't going to be a good idea if multiple deployments share a service account in the same namespace, they'll end up with clashing or identical resources backing them